### PR TITLE
[fix](Nereids): add information function and fix bugs in schemaScan

### PR DESCRIPTION
**Pull request by keanji-x**
# Proposed changes

1. Add information function
```
Database()
User()
Current_User()
Connection_id()
```
2. Fix bugs in schemaScan

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.52 seconds
 load time: 641 seconds
 storage size: 17123011320 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230104113450_clickbench_pr_73512.html

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ConnectionId.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ */
+public class ConnectionId extends ScalarFunction
+        implements LeafExpression, ExplicitlyCastableSignature, AlwaysNotNullable {
+
+    public static final List<FunctionSignature> SIGNATURES = ImmutableList.of(
+            FunctionSignature.ret(VarcharType.SYSTEM_DEFAULT).args()
+    );
+
+    public ConnectionId() {
+        super("connection_id", new ArrayList<>());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ConnectionId.java:43]:
```suggestion
        super("connection_id", ImmutableList.of());
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/CurrentUser.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ */
+public class CurrentUser extends ScalarFunction
+        implements LeafExpression, ExplicitlyCastableSignature, AlwaysNotNullable {
+
+    public static final List<FunctionSignature> SIGNATURES = ImmutableList.of(
+            FunctionSignature.ret(BigIntType.INSTANCE).args()
+    );
+
+    public CurrentUser() {
+        super("current_user", new ArrayList<>());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/CurrentUser.java:43]: ditto

File: regression-test/suites/nereids_syntax_p0/information_schema.groovy [hunk 3]
```diff
@@ -19,4 +19,14 @@ suite("information_schema") {
     List<List<Object>> table =  sql """ select * from information_schema.backends; """
     assertTrue(table.size() > 0) // row should > 0
     assertTrue(table[0].size == 23) // column should be 23
+    List<List<Object>> table =  sql """ select * from information_schema.backends; """
```
Comment (morrySnow) [inline regression-test/suites/nereids_syntax_p0/information_schema.groovy:22]: why repeat L19 here?

File: fe/fe-core/src/main/java/org/apache/doris/catalog/BuiltinScalarFunctions.java [hunk 4]
```diff
@@ -321,6 +325,10 @@ public class BuiltinScalarFunctions implements FunctionHelper {
             scalar(CurrentTime.class, "current_time"),
             scalar(CurrentTimestamp.class, "current_timestamp"),
             scalar(Curtime.class, "curtime"),
+            scalar(Database.class, "database"),
+            scalar(CurrentUser.class, "current_user"),
+            scalar(User.class, "user"),
+            scalar(ConnectionId.class, "connection_id"),
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/BuiltinScalarFunctions.java:331]: add function with lexicographical order

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ConnectionId.java (hunk 1 above)
Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ConnectionId.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/CurrentUser.java (hunk 2 above)
Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/CurrentUser.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/catalog/BuiltinScalarFunctions.java (hunk 4 above)
Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/BuiltinScalarFunctions.java]: done

File: regression-test/suites/nereids_syntax_p0/information_schema.groovy (hunk 3 above)
Comment (XieJiann) [reply regression-test/suites/nereids_syntax_p0/information_schema.groovy]: removed

**Review by morrySnow — APPROVED** [review]

**Pull request merged**